zenodo/modules/records/ext.py

```diff
@@ -44,12 +56,36 @@ def __init__(self, app=None):
         if app:
             self.init_app(app)
 
+    @cached_property
+    def resource_types(self):
+        """."""
+        resource_list = []
+        for item in itervalues(ObjectType.index_id):
+            internal_id = item['internal_id']
+            parent = item.get('parent')
+            if parent:
+                resolved_parent = ObjectType.get(item['internal_id'])['parent']
+                main_type = resolved_parent['title']['en']
+            elif ObjectType.get_subtypes(internal_id):
+                continue
+            else:
+                main_type = ''
+            resource_list.append({
+                'type': main_type,
+                'title': item['title']['en'],
+                'id': internal_id
+            })
+        resource_list.sort(key=lambda x: x['title'])
+        resource_list.insert(0, {'title': 'N/A'})
```

Comment:
Since this is a heavily UI-related mechanic maybe adding the default/null option in the directive might be better